### PR TITLE
#[152464000] [This PR will test the CRUD operations of challenge 2]

### PR DESCRIPTION
**Pull request by kaguna**
#### What does this PR do?
This PR operates  CRUD (Create, Read, Update and Delete) operations to the categories and the recipes.

#### Description of Task to be completed?
It implements the Test Driven Development to the OOP and the data structures in use i.e. list.

#### How should this be manually tested?
To test manually, install the virtual environment  and then use pip to get the necessary apps for testing then store them in the requirements.txt

#### What are the relevant pivotal tracker stories?
[Delivers #152464000](https://www.pivotaltracker.com/story/show/152464000)
[Finishes #152183815](https://www.pivotaltracker.com/story/show/152183815)
[Finishes #152183827](https://www.pivotaltracker.com/story/show/152183827)
[Finishes #152183845](https://www.pivotaltracker.com/story/show/152183845)
[Finishes #152183852](https://www.pivotaltracker.com/story/show/152183852)

#### Screenshot
crud

### CODE REVIEW

**Comment by Sylvance** [comment]
Nice job. Coverage looks good and its builds well.

**Review by jonathankamau — CHANGES_REQUESTED** [review]
Please use pep8 to check for issues in your code. You have several files that do not have new lines at the end.

**Review by jonathankamau — CHANGES_REQUESTED** [review]
I still see that some of your files are missing a new line at the end of the file. Please correct this.

**Review by jonathankamau — CHANGES_REQUESTED** [review]
You have 263 issues to fix on code climate. Please check on this

**Review by jonathankamau — CHANGES_REQUESTED** [review]
Your travis build is failing and your test coverage has dropped. Please investigate this.

File: classes/users.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        self.password = password
+
+    def register_user(self, username, email, password, confirm_password):
+        """
+        This method will validate and verify the user details before storing
+        """
+        nonspace_username = re.sub(r'\s+', ' ', username).strip()
+        regusername = "[a-zA-Z0-9- .]"
+        regemail = r"([a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)"
+        if re.match(regusername, nonspace_username):
```
Comment (jonathankamau) [inline classes/users.py:22]: From line 22 to 41 those are too many `if` statements :flushed: please see how you can refactor that.

**Pull request merged**